api/client/lib.py

```diff
@@ -1,46 +1,95 @@
+"""Base module for making API requests.
+
+Client, GroClient, CropModel, and BatchClient all build on top of endpoints
+exposed in this module. Helper functions or shims or derivative functionality
+should appear in the client classes rather than here.
+"""
+
 from builtins import map
 from builtins import str
 from api.client import cfg
 import logging
 import requests
-import sys
 import time
-from datetime import datetime
 
 CROP_CALENDAR_METRIC_ID = 2260063
 
+REGION_LEVELS = {
+    'world': 1,
+    'continent': 2,
+    'country': 3,
+    'province': 4,  # Equivalent to state in the United States
+    'district': 5,  # Equivalent to county in the United States
+    'city': 6,
+    'market': 7,
+    'group': 8,  # arbitrary grouping of regions of any level
```

Comment:
@reviewers double-check this. Want to use just "group"? Or "regiongroup" or something else?

This constant isn't being used currently in this PR, but I added it to better document the `descendant_level` param of `get_descendant_regions()`. Can use it later to translate region level names to ids or vice versa

Reply:
set_of_regions maybe? depending on how math-inclined our users are. otherwise "group" is fine too imo.

Reply:
it's really just  "other" ... not even necessarily a set of other regions.   it could be a water basin for example or other region which is not necessarily expressible as a union of admin regions.

Reply:
done